packages/jellyfish-wallet-mnemonic/src/hd_node.ts

```diff
@@ -136,22 +136,36 @@ export class MnemonicHdNodeProvider implements WalletHdNodeProvider<MnemonicHdNo
   /**
    * @param {string[]} words to init MnemonicHdNodeProvider
    * @param {Bip32Options} options
+   * @param {boolean} [validate=false] optionally validate mnemonic words. While BIP39 standard doesn't enforce
+   * validation of words to generate the HD seed, this implementation optionally allow you to validate the BIP39 word
+   * list before generating the MnemonicProviderData.
+   * @throws {Error} if mnemonic sentence checksum invalid, if validate=true
    */
-  static fromWords (words: string[], options: Bip32Options): MnemonicHdNodeProvider {
-    const data = this.wordsToData(words, options)
+  static fromWords (words: string[], options: Bip32Options, validate: boolean = false): MnemonicHdNodeProvider {
+    const data = this.wordsToData(words, options, validate)
     return this.fromData(data, options)
   }
 
   /**
    * @param {string[]} words to convert into MnemonicProviderData
    * @param {Bip32Options} options
+   * @param {boolean} [validate=false] optionally validate mnemonic words. While BIP39 standard doesn't enforce
+   * validation of words to generate the HD seed, this implementation optionally allow you to validate the BIP39 word
+   * list before generating the MnemonicProviderData.
    * @return MnemonicProviderData
+   * @throws {Error} if mnemonic sentence checksum invalid, if validate=true
```

Comment:
It's out of the scope of this PR, but should we create custom errors  instead of relying on the base error class? Something like

```typescript
class MnemonicValidationError extends Error {
    message = "mnemonic sentence checksum invalid"
}
```

so we can use it in later error handling with something like:

```typescript
try {
 // ... do something ...
}catch(err){
    if(err instanceof MnemonicValidationError){
        //  handle it somehow
    }
}
```
```typescript
expect(() => {
        MnemonicHdNodeProvider.fromWords(words, regTestBip32Options, false)
      }).not.toThrow(MnemonicValidationError)
```
This is probably something that needs a lot of discussion from the team and is more of a change in convention than something related to this PR.